drift/lib/src/runtime/query_builder/expressions/datetimes.dart

```diff
@@ -107,3 +148,164 @@ class _StrftimeSingleFieldExpression extends Expression<int?> {
   }
 }
+
+/// DateTime modifier constants
+class DateTimeModifier extends Constant<String> {
+  const DateTimeModifier._(super.value);
+
+  /// The "n days" modifiers simply add the specified amount of time to the date
+  /// and time specified by the arguments to the left. Note that "±NNN months"
+  /// works by rendering the original date into the YYYY-MM-DD format, adding
+  /// the ±NNN to the MM month value, then normalizing the result. Thus, for
+  /// example, the date 2001-03-31 modified by '+1 month' initially yields
+  /// 2001-04-31, but April only has 30 days so the date is normalized to
+  /// 2001-05-01. A similar effect occurs when the original date is February 29
+  /// of a leapyear and the modifier is ±N years where N is not a multiple of
+  /// four.
+  const DateTimeModifier.days(int days) : this._('$days days');
+
+  /// The "n hours" modifiers simply add the specified amount of time to the
+  /// date and time specified by the arguments to the left. Note that "±NNN
+  /// months" works by rendering the original date into the YYYY-MM-DD format,
+  /// adding the ±NNN to the MM month value, then normalizing the result. Thus,
+  /// for example, the date 2001-03-31 modified by '+1 month' initially yields
+  /// 2001-04-31, but April only has 30 days so the date is normalized to
+  /// 2001-05-01. A similar effect occurs when the original date is February 29
+  /// of a leapyear and the modifier is ±N years where N is not a multiple of
+  /// four.
+  DateTimeModifier.hours(int hours) : this._('$hours hours');
+
+  /// The "n minutes" modifiers simply add the specified amount of time to the
+  /// date and time specified by the arguments to the left. Note that "±NNN
+  /// months" works by rendering the original date into the YYYY-MM-DD format,
+  /// adding the ±NNN to the MM month value, then normalizing the result. Thus,
+  /// for example, the date 2001-03-31 modified by '+1 month' initially yields
+  /// 2001-04-31, but April only has 30 days so the date is normalized to
+  /// 2001-05-01. A similar effect occurs when the original date is February 29
+  /// of a leapyear and the modifier is ±N years where N is not a multiple of
+  /// four.
+  DateTimeModifier.minutes(int minutes) : this._('$minutes minutes');
+
+  /// The "n seconds" modifiers simply add the specified amount of time to the
+  /// date and time specified by the arguments to the left. Note that "±NNN
+  /// months" works by rendering the original date into the YYYY-MM-DD format,
+  /// adding the ±NNN to the MM month value, then normalizing the result. Thus,
+  /// for example, the date 2001-03-31 modified by '+1 month' initially yields
+  /// 2001-04-31, but April only has 30 days so the date is normalized to
+  /// 2001-05-01. A similar effect occurs when the original date is February 29
+  /// of a leapyear and the modifier is ±N years where N is not a multiple of
+  /// four.
+  DateTimeModifier.seconds(double seconds) : this._('$seconds seconds');
```

Comment:
Let's use `num` here so that int expressions can be used too.

```suggestion
  DateTimeModifier.seconds(num seconds) : this._('$seconds seconds');
```